spark3/src/main/java/org/apache/iceberg/spark/SparkSessionCatalog.java

```diff
@@ -240,8 +245,38 @@ public void renameTable(Identifier from, Identifier to) throws NoSuchTableExcept
     }
   }
 
+  public boolean isHiveCatalogConfigValid(CaseInsensitiveStringMap options, StringBuilder errorMsg) {
+    String hadoopConfUri = SparkSession.active().sparkContext().conf().contains("spark.hadoop.hive.metastore.uris") ?
+            SparkSession.active().sparkContext().conf().get("spark.hadoop.hive.metastore.uris") : null;
```

Comment:
This isn't correct because it only checks for a Spark-configured value, but there could be a value from hive-site.xml. Instead, this should use a Hadoop conf:

```java
Configuration conf = SparkSession.active().sessionState().newHadoopConf();
String envHiveUri = conf.get("hive.metastore.uris", null);
String catalogHiveUri = options.get("uri");
...
```